tutorials/roostats/FourBinInstructional.py

```diff
@@ -0,0 +1,327 @@
+bInt = ROOT.RooStats.SimpleInterval()
+if doBayesian and wspace.set("poi").getSize() == 1:
+    bInt = bc.GetInterval()
+else:
+    print("Bayesian Calc. only supports on parameter of interest")
+
+# use MCMCCalculator  (takes about 1 min)
+# Want an efficient proposal function, so derive it from covariance
+# matrix of fit
+fit = wspace["model"].fitTo(data, Save=True)
+ph = ROOT.RooStats.ProposalHelper()
+ph.SetVariables(fit.floatParsFinal())
+ph.SetCovMatrix(fit.covarianceMatrix())
+ph.SetUpdateProposalParameters(ROOT.kTRUE)  # auto-create mean vars and add mappings
+ph.SetCacheSize(100)
+pf = ph.GetProposalFunction()
+
+mc = ROOT.RooStats.MCMCCalculator(data, modelConfig)
+mc.SetConfidenceLevel(0.95)
+mc.SetProposalFunction(pf)
+mc.SetNumBurnInSteps(500)  # first N steps to be ignored as burn-in
+mc.SetNumIters(50000)
+mc.SetLeftSideTailFraction(0.5)  # make a central interval
+mcInt = ROOT.RooStats.MCMCInterval()
+if doMCMC:
+    mcInt = mc.GetInterval()
+
+# ----------------------------------
+# Make some plots
+c1 = ROOT.gROOT.Get("c1")
+if not c1:
+    c1 = ROOT.TCanvas("c1")
+
+if doBayesian and doMCMC:
+    c1.Divide(3)
+    c1.cd(1)
+elif doBayesian or doMCMC:
+    c1.Divide(2)
+    c1.cd(1)
+
+lrplot = ROOT.RooStats.LikelihoodIntervalPlot(plInt)
+lrplot.Draw()
+
+if doBayesian and wspace.set("poi").getSize() == 1:
+    c1.cd(2)
+    # the plot takes a long time and print lots of error
+    # using a scan it is better
+    bc.SetScanOfPosterior(20)
+    bplot = bc.GetPosteriorPlot()
+    bplot.Draw()
+
+if doMCMC:
+    if doBayesian and wspace.set("poi").getSize() == 1:
+        c1.cd(3)
+    else:
+        c1.cd(2)
+    mcPlot = ROOT.RooStats.MCMCIntervalPlot(mcInt)
+    mcPlot.Draw()
+
+# ----------------------------------
+# querry intervals
+print(
+    "Profile Likelihood interval on s = [{}, {}]".format(plInt.LowerLimit(wspace["s"]), plInt.UpperLimit(wspace["s"]))
+)
+# Profile Likelihood interval on s = [12.1902, 88.6871]
+
+if doBayesian and wspace.set("poi").getSize() == 1:
+    print("Bayesian interval on s = [{}, {}]".format(bInt.LowerLimit(), bInt.UpperLimit()))
+
+if doFeldmanCousins:
+    print(
+        "Feldman Cousins interval on s = [{}, {}]".format(fcInt.LowerLimit(wspace["s"]), fcInt.UpperLimit(wspace["s"]))
+    )
+    # Feldman Cousins interval on s = [18.75 +/- 2.45, 83.75 +/- 2.45]
+
+if doMCMC:
+    print("MCMC interval on s = [{}, {}]".format(mcInt.LowerLimit(wspace["s"]), mcInt.UpperLimit(wspace["s"])))
+    # MCMC interval on s = [15.7628, 84.7266]
+
+t.Stop()
+t.Print()
+
+c1.SaveAs("FourBinInstructional.png")
+
+# TODO: The MCMCCalculator has to be destructed first. Otherwise, we can get
+# segmentation faults depending on the destruction order, which is random in
+# Python. Probably the issue is that some object has a non-owning pointer to
+# another object, which it uses in its destructor. This should be fixed either
+# in the design of RooStats in C++, or with phythonizations.
+del mc
```

Comment:
```suggestion
# TODO: The calculators have to be destructed first. Otherwise, we can get
# segmentation faults depending on the destruction order, which is random in
# Python. Probably the issue is that some object has a non-owning pointer to
# another object, which it uses in its destructor. This should be fixed either
# in the design of RooStats in C++, or with phythonizations.
del plc
del bc
del mc
```

We need to again delete all the calculator classes here, to avoid the deletion order problem.